internal/incident/incident.go

```diff
@@ -381,33 +382,35 @@ func (i *Incident) evaluateEscalations(ctx context.Context, tx *sqlx.Tx, ev *eve
 	return nil
 }
 
-// notifyContacts executes all the given pending notifications of the current incident.
+// notifyContacts executes all the given pending notifications of the current incident in a go routine.
 // Returns error on database failure or if the provided context is cancelled.
 func (i *Incident) notifyContacts(ctx context.Context, ev *event.Event, notifications []*NotificationEntry) error {
+	g, childCtx := errgroup.WithContext(ctx)
 	for _, notification := range notifications {
-		contact := i.runtimeConfig.Contacts[notification.ContactID]
-
-		if i.notifyContact(contact, ev, notification.ChannelID) != nil {
-			notification.State = NotificationStateFailed
-		} else {
-			notification.State = NotificationStateSent
-		}
+		notification := notification
+
+		g.Go(func() error {
+			contact := i.runtimeConfig.Contacts[notification.ContactID]
+			if i.notifyContact(contact, ev, notification.ChannelID) != nil {
+				notification.State = NotificationStateFailed
+			} else {
+				notification.State = NotificationStateSent
+			}
 
-		notification.SentAt = types.UnixMilli(time.Now())
-		stmt, _ := i.db.BuildUpdateStmt(notification)
-		if _, err := i.db.NamedExecContext(ctx, stmt, notification); err != nil {
-			i.logger.Errorw(
-				"Failed to update contact notified incident history", zap.String("contact", contact.String()),
-				zap.Error(err),
-			)
-		}
+			notification.SentAt = types.UnixMilli(time.Now())
+			stmt, _ := i.db.BuildUpdateStmt(notification)
+			if _, err := i.db.NamedExecContext(childCtx, stmt, notification); err != nil {
+				i.logger.Errorw(
+					"Failed to update contact notified incident history", zap.String("contact", contact.String()),
+					zap.Error(err),
+				)
+			}
 
-		if err := ctx.Err(); err != nil {
-			return err
-		}
+			return childCtx.Err()
+		})
 	}
 
-	return nil
+	return g.Wait()
 }
```

Comment:
But is this change in any way relevant to this PR?

Reply:
The diff in this PR is already big enough, so I'd focus on the relevant parts here, populating the table in the database and reworking the plugin execution so that this is possible.

This looks like a change for its own PR, would make sense with or without the rest of this PR and doesn't seem needed for the rest of this PR.